compilers/concrete-compiler/compiler/include/concretelang/TestLib/TestCircuit.h

```diff
@@ -0,0 +1,186 @@
+// Part of the Concrete Compiler Project, under the BSD3 License with Zama
+// Exceptions. See
+// https://github.com/zama-ai/concrete-compiler-internal/blob/main/LICENSE.txt
+// for license information.
+
+#ifndef CONCRETELANG_TESTLIB_TESTCIRCUIT_H
+#define CONCRETELANG_TESTLIB_TESTCIRCUIT_H
+
+#include "boost/outcome.h"
+#include "concrete-protocol.capnp.h"
+#include "concretelang/ClientLib/ClientLib.h"
+#include "concretelang/Common/Csprng.h"
+#include "concretelang/Common/Error.h"
+#include "concretelang/Common/Keysets.h"
+#include "concretelang/Common/Protocol.h"
+#include "concretelang/Common/Values.h"
+#include "concretelang/ServerLib/ServerLib.h"
+#include "concretelang/Support/CompilerEngine.h"
+#include "tests_tools/keySetCache.h"
+#include "llvm/Support/Path.h"
+#include <filesystem>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <thread>
+
+using concretelang::clientlib::ClientCircuit;
+using concretelang::clientlib::ClientProgram;
+using concretelang::csprng::ConcreteCSPRNG;
+using concretelang::error::Result;
+using concretelang::keysets::Keyset;
+using concretelang::serverlib::ServerCircuit;
+using concretelang::serverlib::ServerProgram;
+using concretelang::values::TransportValue;
+using concretelang::values::Value;
+
+namespace concretelang {
+namespace testlib {
+
+class TestCircuit {
+
+public:
+  static Result<TestCircuit>
+  create(Keyset keyset, Message<concreteprotocol::ProgramInfo> programInfo,
+         std::string sharedLibPath, uint64_t seedMsb, uint64_t seedLsb,
```

Comment:
Not at all essential, but it seems that in some places the seed is passed as a `__uint128_t` and sometimes as MSB/LSB.

Reply:
Hmme I think I followed what existed before here, and I think the difference is that `uint64_t` can be specified as a literal, where `__uint128_t` can not.

AFAICT only `ConcreteCSPRNG` takes `__uint128_t` as inputs, which I did not touch I believe. If important, maybe we can change it to take two `uint64_t` ?

Reply:
Not essential, no.

Reply:
Then I'll leave it like that for now.